src/main/java/fi/otavanopisto/restfulptv/server/organizations/OrganizationEntityUpdater.java

```diff
@@ -90,22 +96,30 @@ public void onOrganizationIdUpdateRequest(@Observes OrganizationIdUpdateRequest
   @Timeout
   public void timeout(Timer timer) {
     if (!stopped) {
-      if (!queue.isEmpty()) {
-        String entityId = queue.iterator().next();
-        if (!queue.remove(entityId)) {
-          logger.warning(String.format("Could not remove %s from queue", entityId));
-        }
-
-        ApiResponse<VmOpenApiOrganization> response = ptvApi.getOrganizationApi().apiOrganizationByIdGet(entityId);
-        if (response.isOk()) {
-          cacheResponse(entityId, response.getResponse());
-        } else {
-          logger.warning(String.format("Service %s caching failed on [%d] %s", entityId, response.getStatus(),
-              response.getMessage()));
+      if (running) {
+        return;
+      }
+      
+      try {
+        running = true;
+        if (!queue.isEmpty()) {
+          String entityId = queue.iterator().next();
+          if (!queue.remove(entityId)) {
+            logger.warning(String.format("Could not remove %s from queue", entityId));
+          }
+
+          ApiResponse<VmOpenApiOrganization> response = ptvApi.getOrganizationApi().apiOrganizationByIdGet(entityId);
+          if (response.isOk()) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)